modules/redisearch.ts

```diff
@@ -540,289 +562,441 @@ export class Redisearch extends Module {
+    /**
+     * The 'WITHSCORES' parameter. If set, we also return the relative internal score of each document.
+     */
     withScores?: boolean,
+    /**
+     * The 'WITHPAYLOADS' parameter. If set, we retrieve optional document payloads (see FT.ADD).
+    */
     withPayloads?: boolean,
+    /**
+     * The 'WITHSORTKEYS' parameter. Only relevant in conjunction with SORTBY . Returns the value of the sorting key, right after the id and score and /or payload if requested.
+     */
     withSortKeys?: boolean,
+    /**
+     * The 'FILTER' parameter.  If set, and numeric_field is defined as a numeric field in FT.CREATE, we will limit results to those having numeric values ranging between min and max. min and max follow ZRANGE syntax, and can be -inf , +inf and use ( for exclusive ranges. 
+     */
     filter?: {
+        /**
+         * The numeric_field argument of the 'FILTER' parameter
+         */
         field: string,
+        /**
+         * The min argument of the 'FILTER' parameter
+         */
         min: number,
+        /**
+         * The max argument of the 'FILTER' parameter
+         */
         max: number
-    },
+    }[],
+    /**
+     * The 'GEOFILTER' parameter. If set, we filter the results to a given radius from lon and lat. Radius is given as a number and units.
+     */
     geoFilter?: {
+        /**
+         * The field of the 'GEOFILTER' parameter
+         */
         field: string,
+        /**
+         * The lon argument of the 'GEOFILTER' parameter
+         */
         lon: number,
+        /**
+         * The lat argument of the 'GEOFILTER' parameter
+         */
         lat: number,
+        /**
+         * The radius argument of the 'GEOFILTER' parameter
+         */
         radius: number,
+        /**
+         * The measurement argument of the 'GEOFILTER' parameter
+         */
         measurement: 'm' | 'km' | 'mi' | 'ft'
     },
-    inKeys?: {
-        num: number,
-        field: string
-    },
-    inFields?: {
-        num: number,
-        field: string
-    },
-    return?: {
-        num: number,
-        fields?: {
-            name: string,
-            as?: string
-        }[]
-    },
+    /**
+     * The 'INKEYS' parameter. If set, we limit the result to a given set of keys specified in the list. the first argument must be the length of the list, and greater than zero.
+     */
+    inKeys?: (string | number)[],
+    /**
+     * The 'INFIELDS' parameter. If set, filter the results to ones appearing only in specific fields of the document, like title or URL.
+     */
+    inFields?: (string | number)[],
+    /**
+     *  The 'RETURN' parameter. Use this keyword to limit which fields from the document are returned.
+     */
+    return?: (string | {
+        /**
+         * The name of the field.
+         */
+        field: string,
+        /** 
+         * The 'AS' parameter following a "field" name, used by index type "JSON".
+        */
+        as?: string,
+    })[],
```

Comment:
Why should it be a string? we should only have 1 way to achieve this..

Reply:
It's either a sting list or an {field, as?} list, or both at the same time;
All of these are correct:
```js
return: ["name", "age"]

return: [{
   field: "$.name",
   as: "name"
}, {
   field: "$.age",
   as: "age"
}]

return: [{
   field: "$.name",
   as: "name"
}, "age"]
```